tests/device_tests/test_msg_signmessage.py

```diff
@@ -206,3 +206,79 @@ def test_signmessage(client, coin_name, path, script_type, address, message, sig
     )
     assert sig.address == address
     assert sig.signature.hex() == signature
+
+
+MESSAGE_LENGTHS = (
+    pytest.param("This is a very long message. " * 16, id="normal_text"),
+    pytest.param("ThisIsAMessageWithoutSpaces" * 16, id="no_spaces"),
+    pytest.param("ThisIsAMessageWithLongWords " * 16, id="long_words"),
+    pytest.param("This\nmessage\nhas\nnewlines\nafter\nevery\nword", id="newlines"),
+    pytest.param("Příšerně žluťoučký kůň úpěl ďábelské ódy. " * 16, id="utf_text"),
+    pytest.param("PříšerněŽluťoučkýKůňÚpělĎábelskéÓdy" * 16, id="utf_nospace"),
+    pytest.param("1\n2\n3\n4\n5\n6", id="single_line_over"),
+)
+
+
+@pytest.mark.skip_t1
+@pytest.mark.parametrize("message", MESSAGE_LENGTHS)
+def test_signmessage_pagination(client, message):
+    message_read = ""
+    message += "End."
+
+    def input_flow():
```

Comment:
Out of scope for this PR but I'm wondering how to generalize this for other tests to use. Would be nice to record fixtures for all the pageable screens in tests that have them.

Reply:
indeed, we'd need a debuglink indication that a screen is paginated, and ideally a page position as well.

relatively easy to implement, but i'm thinking maybe the redesign of ButtonRequests will solve this too?